chapters/functions.tex

```diff
@@ -1880,10 +1880,17 @@ \section{Function Inlining and Event Generation}\label{function-inlining-and-eve
 "GenerateEvents" "=" ( false | true )
 \end{lstlisting}\end{synopsis}
 \begin{semantics}
-Has only an effect within a function declaration
+Has only an effect within a function declaration.
+
+By default, \lstinline!GenerateEvents = false! and expressions in the function body that would normally be event-generating shall not generate events, similar to inlining the function body while wrapping all expressions in \lstinline!noEvent!, see \cref{modelica:noEvent}.
+By specifying \lstinline!GenerateEvents = true!, event-generating expressions in the function body shall generate events as normal, similar to inlining the function body without wrapping all expressions in \lstinline!noEvent!.
+This implies \lstinline!Inline = true! unless overridden by specifying one of the inlining annotations with value \lstinline!true! (in particular, \lstinline!GenerateEvents = true! cannot be combined with \lstinline!Inline = false!).
```

Comment:
It is not clear what "This" is: if it refers to both sentences above or just the last one. How about this?
```suggestion
In particular, \lstinline!GenerateEvents = true! implies \lstinline!Inline = true! unless overridden by specifying one of the inlining annotations with value \lstinline!true! (in particular, \lstinline!GenerateEvents = true! cannot be combined with \lstinline!Inline = false!).
```

Reply:
I see the problem with _this_, but also don't quite like _in particular_ as the way of connecting the two sentences.  To me, _in particular_ makes it sound as if the second sentence is a consequence of the first one, rather than a separate statement.  I'd therefore suggest:
```suggestion
To simplify implementation, \lstinline!GenerateEvents = true! implies \lstinline!Inline = true! unless overridden by specifying one of the inlining annotations with value \lstinline!true! (in particular, \lstinline!GenerateEvents = true! cannot be combined with \lstinline!Inline = false!).```

Reply:
@henrikt-ma, what are ``` at the end of your suggestions for? Looks like GitHub markup.

I wasn't sure how to connect the two sentences because I wasn't sure what *this* in the original sentence was referring to. It looked to me that the explanation presented in the sentence before this one was the one that implied that GenerateEvents = true should imply that.... So, *in particular* or *thus* or *therefore* would have fit with that idea. I am also not sure about *To simplify implementation*. To me it suggests that if you don't have an implementation that's what you should do. I thought the agreement was that GenerateEvents = true, a tool will try to inline to be able to generate the events. If inlining is not possible then the tool either tries a more sophisticated algorithm if it has one or issues a diagnostic. With *To simplify implementation* it looks to me more like if you get GenerateEvents = true and you have a sophisticated implementation without inlining, you do not need to try to inline it. I don't know if this distinction matters. Probably the end result will be the same if you generate events by inlining or by some other means. But I want to bring it up to confirm that it's acceptable and is consistent with the group agreement.

Reply:
Hm... I don't see why we need to mention implementations here, or need to use 'This' to reference something.
So what about just saying the minimum: "Having \lstinline!GenerateEvents = true! implies \lstinline!Inline = true! ..."

Reply:
Yes

Reply:
Done now.

Reply:
> @henrikt-ma, what are ``` at the end of your suggestions for? Looks like GitHub markup.

(This was due to a missing newline at the end of the suggestion.  Fixed now.)